third_party/qpg_sdk/qpg6100_sdk.gni

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("//build_overrides/jlink.gni")
+import("//build_overrides/mbedtls.gni")
+import("//build_overrides/qpg6100_sdk.gni")
+
+#import("//build_overrides/openthread.gni")
+
+declare_args() {
+  # Location of the QPG6100 SDK.
+  qpg6100_sdk_root = "${chip_root}/third_party/qpg_sdk/repo"
+}
+
+qpg6100_sdk_root_env = getenv("QPG6100_SDK_ROOT")
```

Comment:
Could we just remove this getenv since the SDK is in third_party?

Reply:
It was left it in as override option of the files during development or preliminary drops. Does this interfere with normal operation of the build system if it's unused ?

Reply:
We should prefer build arguments to override, not environment variables.

Environment variables lead introduce a nondeterminism issue where incremental builds don't pick up new values. Build arguments don't have this problem.

Even if you remove this, we can override by passing the argument `qpg6100_sdk_root` via build arguments.

Reply:
We based off the other integrations, who all use an env variable for their SDK locations. These constructions are best avoided indeed.
I wasn't aware ninja could overrule as well cmdline - will update that in another PR